### PR TITLE
Add 'clean' and 'dependencies' target to makefiles

### PR DESCRIPTION
**Pull request by Phrohdoh**

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -1,5 +1,11 @@
 default:
-	cake -target=default
+		cake -target=default
 
 deps:
-	cake -target=deps
+		cake -target=deps
+
+dependencies:
```
Comment (chrisforbes) [inline makefile]: You should really have a `.PHONY` rule that lists these targets, so make doesn't think these rules produce a file of the target name.

Comment (Phrohdoh) [inline makefile]: Ok, I'll add that to this PR later.

**Comment by Phrohdoh** [comment]
I've added the `.PHONY` rule.

**Comment by abcdefg30** [comment]
You didn't update `build.cake` (`Error: The target 'clean' was not found.`)?

**Comment by Phrohdoh** [comment]
Updated again. I forgot to actually include `clean` in this script.

**Comment by abcdefg30** [comment]
Just for the record: We still want to also clean the dependencies.

**Comment by Phrohdoh** [comment]
This now removes `OpenRA.Mods.RA2/dependencies/*.(exe|dll)`.

**Comment by abcdefg30** [comment]
You need to add the redirect for `dependencies` -> `deps` in windows. At the moment I get "Error: The target 'dependencies' was not found.".

**Comment by Phrohdoh** [comment]
I am really off of my game as of late.

**Comment by Phrohdoh** [comment]
I've updated this:

* Fixed a syntax error in make.ps1

**Comment by abcdefg30** [comment]
Looks good to me now. :+1: / :white_check_mark: Thanks.

**Pull request merged**